core-bundle/src/Image/ImageFactory.php

```diff
@@ -232,6 +249,22 @@ private function createConfig($size, ImageInterface $image): array
         return [$config, $this->getImportantPartFromLegacyMode($image, $size[2])];
     }
 
+    /**
+     * Enhances the resize config by image size settings.
+     */
+    private function enhanceResizeConfig(ResizeConfigurationInterface $config, array $imageSize): void
+    {
+        $config
+            ->setWidth((int) $imageSize['width'])
```

Comment:
We should check if `$imageSize['width']` is set here I think.
```suggestion
            ->setWidth((int) ($imageSize['width'] ?? 0))
```